### DISCUSSION TITLE
add the result of an async function to all templates

### DISCUSSION THREAD

**Discussion by euri10** [encode/starlette#2046] [Q&A]
Use case is the following:
I have 40ish templates, all `{% extends "base.html" %}`
I get a function that does:
```
async def get_x():
    ...fetch db call...
    x = db_call_result
    return x
````
I want in all templates that x value from get_x

1st solution is to pass {"x": await_get(x)} to all templates contexts, not impossible but boring.

2nd solution would have been to use context_processor implemented recently by @alex-oleshkevich but that doesn't support async functions : it would have been elegant to just add a async processor and use `{{ x }}` in base.html

there might be a "hackish" way to do it using the request.session, the way flashes can be implemented, but so far I failed at it

maybe the closed PR here would be a good first step, (https://github.com/encode/starlette/pull/1701)

in any case I'm interested as to know if someone would have ideas to get that x dislayed without touching all the templates :)

**Comment by alex-oleshkevich** [comment]
What about middleware and request.state? This is a pattern i use often.

**Reply by euri10** [reply]
I finally managed to do it with a fastapi global dependency that sets the session to the value retrieved, will check on a middleware, why havn't I thought about it !
